[quant][pt2e] Propagate get_attr meta through known ops only

sxu opened the pull request:
Summary: Avoid situation where the graph traversal finds a matmul node with a `get_attr` as its `args[0]`, and incorrectly propagate the `get_attr`'s meta to everything downstream.

Test Plan: CI

Differential Revision: D56219120

jerryzh168 reviewed (approved):
could you add a test in https://github.com/pytorch/pytorch/blob/main/test/quantization/pt2e/test_metadata_porting.py#L63

sxu committed:
[quant][pt2e] Propagate get_attr meta through known ops only (#124415)

Summary:

Avoid situation where the graph traversal finds a matmul node with a `get_attr` as its `args[0]`, and incorrectly propagate the `get_attr`'s meta to everything downstream.

Test Plan: CI

Reviewed By: jerryzh168

Differential Revision: D56219120
